Frameworks/UIKit/NSString+UIKitAdditions.mm

```diff
@@ -579,158 +579,84 @@ - (CGSize)drawAtPoint:(CGPoint)pt
     return fontExtent;
 }
 
-/**
- @Status Interoperable
-*/
-- (CGSize)sizeWithFont:(UIFont*)font constrainedToSize:(CGSize)size lineBreakMode:(UILineBreakMode)lineBreakMode {
-    CGSize ret;
-    std::wstring wideBuffer = Strings::NarrowToWide<std::wstring>(self);
-    WORD* str = (WORD*)wideBuffer.c_str();
+// Private helper for sizeWith... functions
+// Returns the bounding box size this string would occupy when drawn as specified
+// All sizeWith... functions in this file funnel to this
+- (CGSize)_sizeWithAttributes:(NSDictionary<NSString*, id>*)attributes constrainedToSize:(CGSize)size {
+    NSAttributedString* attributedSelf = [[[NSAttributedString alloc] initWithString:self attributes:attributes] autorelease];
 
-    CGRect rct;
-    rct.origin.x = 0;
-    rct.origin.y = 0;
-    rct.size = size;
+    CTFramesetterRef framesetter = CTFramesetterCreateWithAttributedString((__bridge CFAttributedStringRef)attributedSelf);
+    CFAutorelease(framesetter);
 
-    drawString(font, NULL, str, [self length], rct, lineBreakMode, UITextAlignmentLeft, &ret);
+    return CTFramesetterSuggestFrameSizeWithConstraints(framesetter, CFRangeMake(0, self.length), nullptr, size, nullptr);
+}
 
+// Private helper that converts a UILineBreakMode -> NSParagraphStyle
+// TODO #: NS/CT ParagraphStyle are not properly bridged, and ParagraphStyle is not currently read anywhere
+static inline NSParagraphStyle* _paragraphStyleWithLineBreakMode(UILineBreakMode lineBreakMode) {
+    NSMutableParagraphStyle* ret = [NSMutableParagraphStyle new];
+    ret.lineBreakMode = lineBreakMode;
     return ret;
 }
 
 /**
- @Status Interoperable
+ @Status Caveat
+ @Notes lineBreakMode is currently not fully supported
```

Comment:
I filed this as #1108, I'll add a comment to the task calling out these functions specifically.

---
In reply to: [82511650](https://github.com/Microsoft/WinObjC/pull/1107#discussion_r82511650) [](ancestors = 82511650)